proposals/0030-markdoc.md

```diff
@@ -0,0 +1,295 @@
+		post-3.mdoc
+...
+```
+
+Then, users can query entry frontmatter with the same `getCollection()` and `getEntryBySlug()` APIs:
+
+```astro
+---
+import { getCollection, getEntryBySlug } from 'astro:content';
+
+const blog = await getCollection('blog');
+const firstEntry = await getEntryBySlug('blog', 'post-1');
+---
+```
+
+Users are also free to render Markdoc contents using a `Content` component, configuring Markdoc tags and component mappings as props:
+
+```astro
+---
+import Title from '../components/Title.astro';
+import Marquee from '../components/Marquee.astro';
+import { getEntryBySlug } from 'astro:content';
+
+const mdocEntry = await getEntryBySlug('blog', 'test');
+const { Content } = await mdocEntry.render();
+---
+
+<html lang="en">
+	<body>
+		<Content
+			config={{
+				variables: { underlineTitle: true },
+			}}
+			components={{
+				h1: Title,
+				marquee: Marquee,
+			}}
+		/>
+	</body>
+</html>
+```
+
+See [the detailed API section](#api) for information on all supported props, shared config, and advanced component mapping.
+
+# Background & Motivation
+
+Include any useful background detail that that explains why this RFC is important.
+What are the problems that this RFC sets out to solve? Why now? Be brief!
+
+It can be useful to illustrate your RFC as a user problem in this section.
+(ex: "Users have reported that it is difficult to do X in Astro today.")
+
+# Goals
+
+- **Create an `@astrojs/markdoc` integration** that adds `.mdoc` support to content collections.
+- **Support Astro components and server-rendered UI components** (React, Vue, Svelte, etc) within Markdoc files. Note this leaves client-rendered UI components out-of-scope (see non-goals).
+- **Benchmark Markdoc performance against Markdown and MDX** at 1k and 10k documents. This addresses problem (1) from the previous section.
+
+# Non-goals
+
+- Non-goal: **ESM import and `src/pages/` support** for Markdoc files. [See discussion](https://github.com/withastro/roadmap/discussions/478#discussioncomment-4975072) for context.
+- Out-of-scope: **Allowing the `.md` extension.** This would mean overriding Astro's `.md` renderer, which is tightly coupled to remark and your `markdown` configuration options today. We agree using `.md` for Markdoc is a common use case, and deserves a separate proposal to make Astro's Markdown rendering flexible.
+- Future: **A solution for client-rendered UI components.** Unlike MDX, Markdoc doesn't have a concept of directives, and our compiler doesn't have a clear way to dynamically render client-side components ([see challenges](https://gist.github.com/bholmesdev/491fa440efa8bd73d410d66c0c2143c2)). We will recommend users wrap their components in an Astro component to apply the `client:` directive.
+- Future: **Full alignment with Markdown and MDX** rendered result. Namely, the computed `headings` property (which can be tackled in future releases) and frontmatter manipulation via remark (since remark is incompatible with Markdoc).
```

Comment:
nit: I think we're undecided if remark support for Markdoc is something we actually want (I'm at least somewhere between "unsure" and "against"). Wouldn't want to imply that this is decided as wanted for the future.

Reply:
Ah, I noted this since Remark is fundamentally not compatible with Markdoc right now. This would require changes to Markdoc core to implement support. I could change the language to imply we _could_ explore this with the Markdoc team.